org.metaborg.spoofax.core/src/main/java/org/metaborg/spoofax/core/stratego/StrategoCommon.java

```diff
@@ -62,7 +64,7 @@
         String strategy) throws MetaborgException {
         List<MetaborgException> exceptions = Lists.newArrayList();
         for(ILanguageComponent component : impl.components()) {
-            if(component.facet(StrategoRuntimeFacet.class) == null) {
+            if(component.facet(DynamicClassLoadingFacet.class) == null) {
```

Comment:
This method and the next are very much the same. Can you refactor out the commonalities?
Also, I think this method should call `invoke(ILanguageComponent, IContext, IStrategoTerm,
String)` instead.

Reply:
Refactoring commonalities would probably require a lambda that gets the runtime or something like that.
Also calling the suggested invoke instead is not really possible as it will return `null` both on not having a `DynamicClassLoadingFacet` and when the `invoke(HybridInterpreter, IStrategoTerm, String)` returns `null`. Those `null`s have different effects in the current method (the first does the `continue` of the loop, the second returns the `null`).